### PR TITLE
Unit Testing setup

### PR DESCRIPTION
**Pull request by nmilcoff**
Partially addresses #42.

- Added the existing UT project back to the solution
- Removed old tests and updated dependencies
- Created two first unit tests

### CODE REVIEW

**Comment by nmilcoff** [comment]
True! I created a factory for `WebApiClient` for two reasons mainly:
- It is the same approach Refit uses - so I guess that will make it easier for developers to understand how to start.
- I added the `IRefitService` dependencty as a constructor parameter (for testing easiness) and since we don't want our users to create that by theirselves, the factory just makes sense.

Re: Moq and XUnit, both are open source and popular frameworks, so I think we're good to go with them.

Re: squashing, rebasing and merging, I would suggest we use `rebase and merge` when commits are well isolated and there's not much noise, and `squash and merge` in other cases. Opinions? (cc: @LukasThijs)

**Comment by LukasThijs** [comment]
> True! I created a factory for `WebApiClient` for two reasons mainly:
> 
> * It is the same approach Refit uses - so I guess that will make it easier for developers to understand how to start.
> * I added the `IRefitService` dependencty as a constructor parameter (for testing easiness) and since we don't want our users to create that by theirselves, the factory just makes sense.
> 
> Re: Moq and XUnit, both are open source and popular frameworks, so I think we're good to go with them.
> 
> Re: squashing, rebasing and merging, I would suggest we use `rebase and merge` when commits are well isolated and there's not much noise, and `squash and merge` in other cases. Opinions? (cc: @LukasThijs)

@nmilcoff squash and merge should be sufficient in most cases. This what I primarily encounter and use. Don't really use rebase often so I prefer the second.

File: src/Xablu.WebApiClient.sln [hunk 1]
```diff
@@ -56,30 +58,30 @@ Global
 		{E483FD4E-1575-4F90-91ED-5138FF401EE4}.Release|x64.Build.0 = Release|Any CPU
 		{E483FD4E-1575-4F90-91ED-5138FF401EE4}.Release|x86.ActiveCfg = Release|Any CPU
 		{E483FD4E-1575-4F90-91ED-5138FF401EE4}.Release|x86.Build.0 = Release|Any CPU
-		{9B025323-1C9F-42EC-B26B-5C1739BDF748}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
```
[the remark this answers is not in the document]
Comment (nmilcoff) [reply src/Xablu.WebApiClient.sln]: I think this is an intended change in this case, I think during the merge from `Implement-Refit` to `develop` our .sln file got corrupted somehow

[the remark this answers is not in the document]
Comment (LukasThijs) [reply src/Xablu.WebApiClient.sln]: Ok when I merged I reverted these changes and it still worked, but lets keep it this way. If it occurs again we should take better look into it.

**Pull request merged**